content-src/components/DiscoveryStreamComponents/DSImage/DSImage.jsx

```diff
@@ -0,0 +1,107 @@
+import {cache} from "./cache";
+import React from "react";
+import ReactDOM from "react-dom";
+
+export class DSImage extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.onOptimizedImageError = this.onOptimizedImageError.bind(this);
+
+    this.state = {
+      isSeen: false,
+      optimizedImageFailed: false,
+    };
+  }
+
+  onSeen(element) {
+    if (this.state && element[0].isIntersecting) {
```

Comment:
Yeah still seeing issues with the [0] This event can fire multiple intersections, and the first one might not be the most up to date.

I see about a 50/50 chance an image shows as is.

I fixed this by trying this:

```
  onSeen(entries) {
    if (this.state) {
      if (entries.some(entry => entry.isIntersecting)) {
        if (this.props.optimize) {
          this.setState({
            containerWidth: ReactDOM.findDOMNode(this).clientWidth,
          });
        }

        this.setState({
          isSeen: true,
        });

        // Stop observing since element has been seen
        this.observer.unobserve(ReactDOM.findDOMNode(this));
      }
    }
  }
```

It kinda mentions this here but tbh they don't do enough to stress this: https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API#Intersection_change_callbacks